scripts/treedb_kahip_partition.py

```
@@ -0,0 +1,104 @@
+            continue
+        algorithm, encoded = digest.split("=", 1)
+        if algorithm != "sha256" or not size.isdecimal():
+            raise SystemExit("unsupported KaHIP RECORD entry")
+        payload = distribution.locate_file(path).read_bytes()
+        expected = base64.urlsafe_b64decode(encoded + "=" * (-len(encoded) % 4))
+        if len(payload) != int(size) or hashlib.sha256(payload).digest() != expected:
+            raise SystemExit("KaHIP RECORD payload integrity mismatch")
+
+
+if len(sys.argv) != 3:
+    raise SystemExit("usage: treedb_kahip_partition.py INPUT OUTPUT")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)